cmd/storage-rest-server.go

```diff
@@ -60,9 +66,22 @@ func (s *storageRESTServer) IsValid(w http.ResponseWriter, r *http.Request) bool
 		s.writeErrorResponse(w, fmt.Errorf("client time %v is too apart with server time %v", requestTime, utcNow))
 		return false
 	}
+	instanceID := r.URL.Query().Get(storageRESTInstanceID)
+	if instanceID != s.instanceID {
+		// Let minio will re-initiate fresh client and validate format.json again
+		s.writeErrorResponse(w, errConnectionStale)
+		return false
+	}
 	return true
 }
 
+// GetInstanceID - returns the instance ID of the server.
+func (s *storageRESTServer) GetInstanceID(w http.ResponseWriter, r *http.Request) {
+	// Skip s.IsValid() check as the request will not have the instanceID
```

Comment:
This is an anonymous call? - isn't it a bad idea some malicious server can trick this storage server to be a legitimate server. I would suggest we validate the JWT nonetheless but don't have to validate instanceID if its not set yet.